### PR TITLE
Test PureLocalizationTrimmer with actual pose graph

### PR DESCRIPTION
**Pull request by gaschler**

### CODE REVIEW

File: cartographer/mapping/internal/3d/pose_graph_3d_test.cc [hunk 1]
```diff
@@ -51,16 +65,35 @@ class PoseGraph3DTest : public ::testing::Test {
...
+        common::make_unique<pose_graph::FakeOptimizationProblem3D>();
+    optimization_problem_ = optimization_problem.get();
+    pose_graph_ = common::make_unique<PoseGraph3DForTesting>(
+        pose_graph_options_, std::move(optimization_problem),
+        thread_pool_.get());
   }
 
   proto::PoseGraphOptions pose_graph_options_;
   std::unique_ptr<common::ThreadPool> thread_pool_;
+  pose_graph::OptimizationProblem3D* optimization_problem_;
```
Comment (pifon2a) [inline cartographer/mapping/internal/3d/pose_graph_3d_test.cc]: You do not use `optimization_problem_` anywhere? Will you need it in future?

Comment (gaschler) [reply cartographer/mapping/internal/3d/pose_graph_3d_test.cc]: Removed. I first tried to mock it for this test and this is the left over.

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**